### PR TITLE
Code Cleanup

### PR DESCRIPTION
**Pull request by kumar-ab**
Decluttered the repository to remove those outdated apps in which the code was breaking.
Modified few apps to ensure that the basic apps for PIT, CIT & GST microsimulation are now up and running

### CODE REVIEW

**Comment by martinholmer** [comment]
@kumar-ab said in PR #179:

> Decluttered the repository to remove those outdated apps in which the code was breaking.
> Modified few apps to ensure that the basic apps for PIT, CIT & GST microsimulation are now up and running

Great that you are doing this cleanup!

About the cryptic Travis error:
Will the error go away if you remove the `--pep8` off the end of the coverage command?

Screen Shot 2020-12-15 at 9 35 39 AM

**Comment by jdebacker** [comment]
@martinholmer is right about removing the `--pep8`.

If you want to continue with the checks for formatting, you'll want to update to `pycodestyle`.  To do this, replace

`pip install pep8` with `pip install pytest-pycodestyle`

and 

`python -c "import taxcalc"; coverage run -m pytest -v -m "not  local" --pep8` with

`python -c "import taxcalc"; coverage run -m pytest -v -m "not  local" --pycodestyle`

**Comment by kumar-ab** [comment]
Thanks @martinholmer and @jdebacker.
I updated .travis.yml as suggested by @jdebacker. The pep8 error is now gone. However, not two of the tests in test_utils failed (test_create_distribution_tables & test_dist_table_sum_row). On my computer pytest and pycodestyle still don't throw an error! I am not able to debug this. I am think this has something to do with the NaN coming as column header in the table below.
Capture

What am I missing?

**Comment by jdebacker** [comment]
@kumar-ab - great progress!

My guess is that this is an issue with changes in a newer version of the Pandas package.  If you do `conda list pandas` in your terminal (or Anaconda prompt) from within the Python environment you are working with, what version is reported?

**Comment by martinholmer** [comment]
@kumar-ab said:

> Thanks @martinholmer and @jdebacker.
> I updated .travis.yml as suggested by @jdebacker. The pep8 error is now gone. However, two of the tests in test_utils failed (test_create_distribution_tables & test_dist_table_sum_row). On my computer pytest and pycodestyle still runs without an error! I am not able to debug this. 

Your Travis job loaded the current version of pandas, which is 1.1.5.
What @jdebacker is saying is that it is very likely that you have a much older version installed on your computer.
You will be better able to debug the problems if you upgrade your computer to use pandas 1.1.5. 
Then you should get the same errors on your computer as you see happening in the Travis run.
Does this make sense?

**Comment by jdebacker** [comment]
@kumar-ab - please see my PR to your branch. Try that locally w an updated Pandas.

**Comment by kumar-ab** [comment]
@jdebacker I have merged your PR on my branch and have updated the pandas to 1.1.5 (in fact, I've updated all the packages). Still no success! Now the error in the Build seems to be with the line **conda env update -f environment.yml**

Just to add the pytest still passes without any error on my computer and the code also runs perfectly fine.

File: environment.yml [hunk 1]
```diff
@@ -1,8 +1,8 @@
-name: taxcalc-dev
+name: taxcalc-india-dev
```
Comment (jdebacker) [inline environment.yml:1]: @kumar-ab You changed the name of the environment here.  So you'll need to change it in `.travis.yml` in the line
```
source activate taxcalc-dev
```
That might fix the latest error for you.

**Comment by kumar-ab** [comment]
The name change of the environment is strange as I did not make any change on the environment.yml file!

As suggested by @jdebacker I have made necessary changes in the .travis.yml file. The PR reflects "no conflicts with the base branch" now (however, The Build did not run). Does that mean the PR is now ready to be merged? If that is so, please merge the PR.
Regards

**Comment by jdebacker** [comment]
@kumar-ab says:

> The name change of the environment is strange as I did not make any change on the environment.yml file!

Oops -this probably got into my PR to your branch without me noticing.  I had changed the name to the environment to avoid conflicts with the `taxcalc` environment I was using with the US calculator...  I'll start a review to change back.

File: .travis.yml [hunk 2]
```diff
@@ -11,19 +12,23 @@ install:
   - export PATH="$HOME/miniconda/bin:$PATH"
   - conda config --set always_yes yes --set changeps1 no
   - conda update conda
-  - conda create -n taxcalc-dev python=$TRAVIS_PYTHON_VERSION;
-  - source activate taxcalc-dev
+  - conda create -n taxcalc-india-dev python=$TRAVIS_PYTHON_VERSION;
```
Comment (jdebacker) [inline .travis.yml:15]: Can change lines 14 and 15 back to:
```
conda create -n taxcalc-dev python=$TRAVIS_PYTHON_VERSION;
source activate taxcalc-dev
```

File: environment.yml (hunk 1 above)
Comment (jdebacker) [inline environment.yml:1]: Can change back to `taxcalc-dev`.

File: taxcalc/tests/test_4package.py [hunk 3]
```diff
@@ -20,7 +20,7 @@ def test_for_package_existence():
     the local mark prevents test from running on GitHub.
     """
     out = subprocess.check_output(['conda', 'list', 'taxcalc']).decode('ascii')
-    envless_out = out.replace('taxcalc-dev', 'environment')
+    envless_out = out.replace('taxcalc-india-dev', 'environment')
```
Comment (jdebacker) [inline taxcalc/tests/test_4package.py:23]: Can change back to `taxcalc-dev`

**Review by jdebacker — COMMENTED** [review]
Places to change environment name back to `taxcalc-dev`, if you wish.

**Comment by jdebacker** [comment]
@kumar-ab writes:

> The PR reflects "no conflicts with the base branch" now (however, The Build did not run). Does that mean the PR is now ready to be merged?

No!  Travis-CI.org moved over to Travis-CI.com and now charges for runs after some limit.  I think I set up Travis-CI here and my free runs are maxed out, so the continuous integration testing isn't running any more ([see FAQs on billing](https://docs.travis-ci.com/user/billing-faq/) if interested).  But we can set up a GitHub Action to do our testing (I have done this in other repos, it is free and has many more features available).  I will set this up here.

**Comment by jdebacker** [comment]
@kumar-ab I just opened a PR to your branch with the GH Action.  I assumes you change the name of the environment back to `taxcalc-dev`.

**Comment by kumar-ab** [comment]
Ah! such a relief to finally see "All checks have passed" 🥇

**Pull request merged**